packages/UI/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,80 @@
+import * as Dialog from '@radix-ui/react-dialog';
+import { X } from 'lucide-react';
+import { tv } from 'tailwind-variants';
+export { Close, Description } from '@radix-ui/react-dialog';
+
+export interface ModalProps {
+  /**
+   * hasCloseBtn: boolean - defaulted true, will render a close button in the top right corner of the modal
+   */
+  hasCloseBtn?: boolean;
+  /**
+   * trigger: React.ReactNode - the element that will trigger the modal to open
+   */
+  trigger: React.ReactNode;
+  /**
+   * header: React.ReactNode - the element that will be rendered in the top of the modal
+   */
+  header: React.ReactNode;
+  /**
+   * content: React.ReactNode - the element that will be rendered in the middle of the modal
+   */
+  content: React.ReactNode;
+  /**
+   * footer: React.ReactNode - the element that will be rendered at the bottom of the modal
+   */
+  footer: React.ReactNode;
```

Comment:
let's use the same description for these props from https://www.radix-ui.com/primitives/docs/components/dialog#root